### ISSUE TITLE
Does not warn me on `raise` with implicit `try`

### ISSUE THREAD

**Issue by sobolevn** [rrrene/credo#398]
### Environment

* Credo version (`mix credo -v`): 0.8.1
* Erlang/Elixir version (`elixir -v`): 

> Erlang/OTP 19 [erts-8.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
> 
> Elixir 1.4.4

* Operating system: macOS

### What were you trying to do?

This block raises two warnings:

```elixir
defp load_values do
   try do
     :petrovich_elixir
      |> Config.get_env(@setting)
      |> File.read!
      |> Poison.decode!
  rescue
    e in File.Error ->
      raise RulesFileException, message: Exception.message(e)
    end
end
```

First one is about code readability:

>   Code Readability                                                              
> ┃ 
> ┃ [R] ↓ Prefer using an implicit `try` rather than explicit `try`.
> ┃       lib/petrovich/stores/base.ex:41:9 #(Petrovich.Store.__using__.load_values)
> 

It also raises this warning about `reraise`: 

>   Warnings - please take a look                                                 
> ┃ 
> ┃ [W] → Use `reraise` inside a rescue block to preserve the original stacktrace.
> ┃       lib/petrovich/stores/base.ex:48:13 #(Petrovich.Store.__using__.load_values)

But when I try to change it to the implicit version:

```elixir
      defp load_values do
        :petrovich_elixir
        |> Config.get_env(@setting)
        |> File.read!
        |> Poison.decode!
      rescue
        e in File.Error ->
          raise RulesFileException, message: Exception.message(e)
      end
```

It does not raise any warnings at all.

### Expected outcome

One warning with `reraise` note.

### Actual outcome

No warnings.

Full source: https://github.com/petrovich/petrovich_elixir/blob/053f38851b605c9016c8252247a73ee2612957e6/lib/petrovich/stores/base.ex#L40

**Comment by rrrene** [comment]
You are right. Thx for reporting! :+1:
